### ISSUE TITLE
Consider making the sender generate url previews, as with e2e thumbnails

### ISSUE THREAD

**Issue by Matrixcoffee** [vector-im/riot-web#3255]
### Description

Consider making the sender generate url previews. We are already trusting clients to faithfully generate thumbnails in e2e encrypted rooms anyway, so this is not a big jump in trust. Furthermore, someone who posts a link to malicious content under their control can already make the url preview say whatever they want, regardless of whether it is generated by a homeserver or sent directly.

On the plus side, it improves privacy, especially for single-user homeservers. It also improves
efficiency, as the site is only accessed once, rather than by every homeserver in the room, and reduces load on the servers.
 
### Steps to reproduce

- n/a

Log: not sent    

### Version information
n/a

- **Platform**: any

For the web app:

- **Browser**: any
- **OS**: any
- **URL**: any

For the desktop app:

- **OS**: any
- **Version**: future

**Comment by ara4n** [comment]
yeah, i'm inclined to agree, although i think we should still let clients have the option of generating their own thumbs if they prefer.

we clearly need a standard API for clientside thumbnailing in general, which links in turn into the whole general "human representation of events" problem.

**Comment by ara4n** [comment]
This also has the advantage of letting the sender sculpt what the preview looks like in future (as you can do on FB for instance)
